Use larger of PATH_MAX and page size when allocating buffer.
See BZ2418
Note: CVS 1.24 does not build, MAX macro is undefined

Index: glibc-2.3.6/sysdeps/unix/sysv/linux/getcwd.c
===================================================================
--- glibc-2.3.6.orig/sysdeps/unix/sysv/linux/getcwd.c
+++ glibc-2.3.6/sysdeps/unix/sysv/linux/getcwd.c
@@ -86,7 +86,9 @@
 	  return NULL;
 	}
 
-      alloc_size = PATH_MAX;
+      alloc_size = __getpagesize ();
+      if (alloc_size < PATH_MAX)
+        alloc_size = PATH_MAX;
     }
 
   if (buf == NULL)
